third_party/terraform/resources/resource_storage_bucket.go

```diff
@@ -734,7 +734,7 @@ func flattenCors(corsRules []*storage.BucketCors) []map[string]interface{} {
 
 func expandBucketEncryption(configured interface{}) *storage.BucketEncryption {
 	encs := configured.([]interface{})
-	if encs == nil || encs[0] == nil {
+	if encs == nil || len(encs) == 0 || encs[0] == nil {
```

Comment:
Same here. The panics happening in validator were because they were using an outdated TPG schema and supplying bad values if I remember right. I'm hesitant to bandaid over that in upstream.

Reply:
I figured it was better to have consistency between the two resources instead of forking due to bad validator usage. The check is a sane one and returning nil here makes sense. I don't feel there is harm adding an extra assertion upstream in this case.

Reply:
Fair enough! Mind adapting the MM template for these functions in all 3 cases so they're consistent? For example, `len(encs)` will always be `0` because it's `nil` so we can omit that check.

Var names can differ, but that's:

```go
	l := v.([]interface{})
	if len(l) == 0 || l[0] == nil {
		return nil, nil
	}
	raw := l[0]
	original := raw.(map[string]interface{})
	transformed := make(map[string]interface{})
```

Reply:
I'm not sure what you mean by updating the MM template? Do you mean updating the Terraform handwritten resource to look like the validator as done [here](https://github.com/GoogleCloudPlatform/magic-modules/pull/2702/files#diff-ecf24ee803ba41689acef62cd5377a6aR132)

Or do you mean update both of them to look more like the generated MM resources?

Reply:
Both of them to look more like the generated MM resource, sorry! I can see 3 ways it's done in this PR, and all of them can be replaced with the MM-looking way.

Reply:
Gotcha, yeah it's a bit ugly. I'll keep this PR to just catching validator up with the upstream changes. I think refactoring handwritten resources to look more like generated resources is out of scope for this and probably worth doing once these changes are automated instead of made by hand anyway.